flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -1894,6 +1895,52 @@ object TaskManager {
       override def getValue: Double =
         ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
     })
+    
+    // Preprocessing steps for registering cpuLoad
+    // fetch the method to get process CPU load
+    val getCPULoadMethod: Method = getMethodToFetchCPULoad()
+
+    // define the fetchCPULoad method as per the fetched getCPULoadMethod
+    // dummy initialisation
+    var fetchCPULoad:(Any) => Double = (obj:Any) => -1
```

Comment:
We need to define fetchCPULoad method as per the object returned by getMethodToFetchCPULoad. val cannot be reassigned and has to be initialised. Hence var has been used to overcome this.